docs/technical_reference/costing/automate_detailed_unit_costing_rst_file.py

```diff
@@ -0,0 +1,191 @@
+            else:
+                on_list = True
+                line_index = l
+                break
+
+        # if new entry or if overwriting existing entry
+        if (not on_list) or (OVERWRITE and (on_list)):
+
+            # check that entries in OVERWRITE_LIST are valid
+            if OVERWRITE and len(OVERWRITE_LIST):
+                if all(
+                    overwrite_item not in unit_name_list
+                    for overwrite_item in OVERWRITE_LIST
+                ):
+                    raise ValueError(
+                        "Entries provided in OVERWRITE_LIST are invalid.\nOVERWRITE_LIST should be empty or contain a list of valid unit names (e.g., 'ion_exchange')"
+                    )
+
+            # if this is a new entry, append the unit name to the landing page for all unit costing
+            if not on_list:
+                # append unit doc to index
+                with open("detailed_unit_model_costing.rst", "a") as f:
+                    f.write(f"   {unit_name_list[i]}\n")
+            else:
+                # either overwrite all units in the landing page OR overwrite specific units specified on OVERWRITE_LIST
+                if (not len(OVERWRITE_LIST)) or (
+                    any(item == unit_name_list[i] for item in OVERWRITE_LIST)
+                ):
+                    lines[line_index] = f"   {unit_name_list[i]}\n"
+                    with open("detailed_unit_model_costing.rst", "w") as f:
+                        f.writelines(lines)
+                else:
+                    pass
+
+            if (
+                (not on_list)
+                or (not len(OVERWRITE_LIST))
+                or (any(item == unit_name_list[i] for item in OVERWRITE_LIST))
+            ):
+                with open(f"{unit_name_list[i]}.rst", "w") as f:
+
+                    # write doc title based on unit name
+                    if unit_name_list[i] in title_exceptions:
+                        f.write(f"{title_exceptions[unit_name_list[i]]} Costing Method")
+                        f.write("\n")
+                        f.write(
+                            "="
+                            * len(
+                                f"{title_exceptions[unit_name_list[i]]} Costing Method\n"
+                            )
+                        )
+                    else:
+                        f.write(f"{unit_title} Costing Method")
+                        f.write("\n")
+                        f.write("=" * len(f"{unit_title} Costing Method\n"))
+
+                    # TODO: add parameter tables
+                    f.write("\n\nCosting Method Parameters\n")
+                    f.write("+" * len("Costing Method Parameters"))
+                    f.write(
+                        f"\n\nThe following parameters are constructed when applying the `cost_{unit_name_list[i]}` costing method in the ``watertap_costing_package``:\n\n"
```

Comment:
Is it important to clarify in this section that the parameters exist on m.fs.costing.unit (and are used across the flowsheet) while the variables on m.fs.unit.costing (and unique to each unit op)?

Reply:
@hunterbarber do you think this should be duplicated on every costing tech ref, or should it just be in one place in the higher level costing documentation (if not there already)? I could go either way but would lean towards duplicating across all templates so that users don't miss the message.